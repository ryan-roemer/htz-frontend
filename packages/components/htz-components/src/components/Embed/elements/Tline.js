/* *************************************************************** *
 * This element accepts these inputTemplates:
[
com.polobase.TlineEmbed,
]
 * This element does not emits an onLoad event
 * *************************************************************** */

import React from 'react';
import PropTypes from 'prop-types';
import { createComponent, } from 'react-fela';
import Caption from '../../Caption/Caption';
import { appendScript, } from '../../../utils/scriptTools';

const tlineWrapper = () => ({
  textAlign: 'center',
  margin: '0 auto',
});

const TlineWrapper = createComponent(tlineWrapper, 'figure', props =>
  Object.keys(props)
);

export default class Tline extends React.Component {
  static propTypes = {
    /**
     * The settings values extracted from the Tline source code.
     */
    settings: PropTypes.shape({
      src: PropTypes.string.isRequired,
      'data-alias': PropTypes.string.isRequired,
      'data-version': PropTypes.string.isRequired,
    }).isRequired,
    /**
     * Caption for this element (Passes down to the [***Caption***](./#caption) component).
     */
    caption: PropTypes.string,
    /**
     * Credit (Passes, along with the Caption, down to the [***Caption***](./#caption) component).
     */
    credit: PropTypes.string,
  };

  static defaultProps = {
    caption: '',
    credit: '',
  };

  componentDidMount() {
    const src = 'https://tline.io/assets/js/release/loader.js';
    const async = true;
    const id = 'tline-js';

    appendScript(src, id, async);
  }

  render() {
    const settings = this.props.settings;
    const src = settings.src;
    const alias = settings['data-alias'];
    const version = settings['data-version'];

    return (
      <TlineWrapper
        data-alias={alias}
        data-version={version}
        className="tline-iframe op-interactive"
      >
        <iframe
          title="video embed"
          className="tline-iframe-nojs"
          src={src}
          height="700"
          width="100%"
          scrolling="yes"
          frameBorder="0"
        />
        <Caption
          caption={this.props.caption}
          credit={this.props.credit}
        />
      </TlineWrapper>
    );
  }
}
