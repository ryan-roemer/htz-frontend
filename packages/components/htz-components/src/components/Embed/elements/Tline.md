<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**

- [Code Example](#Tline-component)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

### Tline component

An example for Tline timeline Embed.

```jsx static
<Tline
  settings = {{
    src: 'https://view.tline.io/embed/donald-trumps-path-to-the-white-house/?noJS=1&v=0.1.1',
    'data-alias': 'donald-trumps-path-to-the-white-house',
    'data-version': '0.1.1',
  }}
  caption = "timeline"
  credit = "Tline"
/>
```
