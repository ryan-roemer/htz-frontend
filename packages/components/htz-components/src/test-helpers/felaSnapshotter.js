import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
