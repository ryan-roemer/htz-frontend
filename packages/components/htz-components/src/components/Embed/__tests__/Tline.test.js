import React from 'react';
import felaSnapshotter from '../../../test-helpers/felaSnapshotter';

import Tline from '../elements/Tline';

it('Tline ', () => {
  const snapshot = felaSnapshotter(
    <Tline
      content="<figure data-alias=\&quot;donald-trumps-path-to-the-white-house\&quot; data-version=\&quot;0.1.1\&quot; class=\&quot;tline-embed op-interactive\&quot; style=\&quot;text-align: center; margin: 0 auto\&quot;><embed class=\&quot;tline-embed-nojs\&quot; src=\&quot;https://view.tline.io/embed/donald-trumps-path-to-the-white-house/?noJS=1&v=0.1.1\&quot; height=\&quot;500\&quot; width=\&quot;500\&quot; scrolling=\&quot;yes\&quot; frameborder=\&quot;0\&quot;></embed></figure>"
      caption="tline"
      credit="tline"
      embedType="tline"
      elementType="embedElement"
      settings={{
        src:
          'https://view.tline.io/embed/donald-trumps-path-to-the-white-house/?noJS=1&v=0.1.1',
        'data-alias': 'donald-trumps-path-to-the-white-house',
        'data-version': '0.1.1',
      }}
      inputTemplate="com.polobase.TlineEmbed"
      contentId="7.5196"
      contentName="tline"
    />
  );
  expect(snapshot).toMatchSnapshot();
});
