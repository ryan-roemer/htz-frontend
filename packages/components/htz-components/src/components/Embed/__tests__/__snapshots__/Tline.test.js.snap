// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tline  1`] = `
Object {
  "component": <div>
    <figure
      className="tline-iframe op-interactive htz-a htz-b"
      data-alias="donald-trumps-path-to-the-white-house"
      data-version="0.1.1"
    >
      <iframe
        className="tline-iframe-nojs"
        frameBorder="0"
        height="700"
        scrolling="yes"
        src="https://view.tline.io/embed/donald-trumps-path-to-the-white-house/?noJS=1&v=0.1.1"
        title="video embed"
        width="100%"
      />
    </figure>
  </div>,
  "styles": ".htz-a{text-align:center}.htz-b{margin:0 auto}",
}
`;
